.github/workflows/minikube.yml

```diff
@@ -0,0 +1,172 @@
+---
+name: MiniKube Heml Deployment
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches: [main]
+
+jobs:
+  job1:
+    runs-on: ubuntu-latest
+    name: build and deploy to minikube
+    steps:
+      - uses: actions/checkout@v2
+
+      # HELM Config
+      - name: Install Helm
+        uses: azure/setup-helm@v3
+        with:
+          version: v3.10.0
+
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v2
+        with:
+          node-version: ${{ matrix.node-version }}
+          cache: 'npm'
+
+      - run: mkdir logs
+
+      # Install dependencies
+      - run: npm ci
+
+      # Create certs
+      - run: sudo chown -R runner:docker certs
+      - run: npm run certs
+
+      # Remove route set for openshift
+      - name: remove route config
+        run: sudo rm helm/bailo/templates/route.yaml
+
+      # Add bailo NodePort
+      - name: Create NodePort Config
```

Comment:
Change this to be in the helm chart, possibly using an `ingress`.

Reply:
For ingress to work (minikube specific), we need;
- Additional Ingress Controller Pod,  simply done by  "$ minikube addons enable ingress".
- Then create an ingress rule, and for rule to work we need dns mapping to IP; this can be done via /etc/hosts.
- **Issue**: However for the mapping we need IP which is dynamic and takes more than few seconds to be assigned. Hence, not able to create mapping in /etc/hosts on fly
- Once /etc/hosts is amended, we have all request coming to minikube cluster handed over to ingress controller, which validates the rules and forwards that request to service.